src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+        public DateTime ToDateTime(TimeOnly time) => new DateTime(_dayNumber * TimeSpan.TicksPerDay + time.Ticks);
+
+        /// <summary>
+        /// Returns a DateTime instance with the specified input kind that is set to the date of this DateOnly instance and the time of specified input time.
+        /// </summary>
+        /// <param name="time">The time of the day.</param>
+        /// <param name="kind">One of the enumeration values that indicates whether ticks specifies a local time, Coordinated Universal Time (UTC), or neither.</param>
+        /// <returns>The DateTime instance composed of the date of the current DateOnly instance and teh time specified by the input time.</returns>
+        public DateTime ToDateTime(TimeOnly time, DateTimeKind kind) => new DateTime(_dayNumber * TimeSpan.TicksPerDay + time.Ticks, kind);
+
+        /// <summary>
+        /// Returns a DateOnly instance that is set to the date part of the specified dateTime.
+        /// </summary>
+        /// <param name="dateTime">The DateTime instance.</param>
+        /// <returns>The DateOnly instance composed of the date part of the specified input time dateTime instance.</returns>
+        public static DateOnly FromDateTime(DateTime dateTime) => new DateOnly(DayNumberFromDateTime(dateTime));
+
+        /// <summary>
+        /// Compares the value of this instance to a specified DateOnly value and returns an integer that indicates whether this instance is earlier than, the same as, or later than the specified DateTime value.
+        /// </summary>
+        /// <param name="value">The object to compare to the current instance.</param>
+        /// <returns>Less than zero if this instance is earlier than value. Greater than zero if this instance is later than value. Zero if this instance is the same as value.</returns>
+        public int CompareTo(DateOnly value)
+        {
+            if (_dayNumber < value._dayNumber) return -1;
+            if (_dayNumber > value._dayNumber) return 1;
+            return 0;
+        }
+
+        /// <summary>
+        /// Compares the value of this instance to a specified object that contains a specified DateOnly value, and returns an integer that indicates whether this instance is earlier than, the same as, or later than the specified DateOnly value.
+        /// </summary>
+        /// <param name="value">A boxed object to compare, or null.</param>
+        /// <returns>Less than zero if this instance is earlier than value. Greater than zero if this instance is later than value. Zero if this instance is the same as value.</returns>
+        public int CompareTo(object? value)
+        {
+            if (value == null) return 1;
+            if (!(value is DateOnly))
+            {
+                throw new ArgumentException(SR.Arg_MustBeDateOnly);
+            }
+
+            return CompareTo((DateOnly)value);
```

Comment:
💡 If the project can use C# 9, this would also work:

```suggestion
            if (value is not DateOnly dateOnly)
            {
                throw new ArgumentException(SR.Arg_MustBeDateOnly);
            }

            return CompareTo(dateOnly);
```